lib/js/jsre.nim

```diff
@@ -44,3 +45,7 @@ func test*(self: RegExp; pattern: cstring): bool {.importjs: "#.test(#)".}
 
 func toString*(self: RegExp): cstring {.importjs: "#.toString()".}
```

Comment:
this should be deprecated as mentioned by @araq in various PR's eg: https://github.com/nim-lang/Nim/pull/16409#discussion_r549346583

* add a `toCstring`
* deprecate this one and make sure it's not used in docs/tests